### PR TITLE
Use GitHub actions

### PR DESCRIPTION
**Pull request by Donatien26**
what do you think of this ?
i recommend to squash commit before merging :)

### CODE REVIEW

**Comment by mboisnard** [comment]
Hi,

Thanks for this PR ! Could you please rebase your commits and improve your commit message? DCO bot failed due to the signed-off rule (please use: git commit -s -m 'Ceci est mon message de validation' to sign your commit with your git configured email address)

Some points in review and below:
- we can remove .travis.yml file now
- we can use the Cache Action to locally cache maven dependencies https://github.com/actions/cache

File: .github/workflows/tag.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
```
Comment (mboisnard) [inline .github/workflows/tag.yaml:16]: We use Java 8 in pom

File: .github/workflows/workflows.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Set current version
+        run: mvn -B versions:set -DnewVersion=${GITHUB_REF##*/}
+      - name: Build with Maven
+        run: mvn -B clean verify
+      - name: Upload jar
```
Comment (mboisnard) [inline .github/workflows/workflows.yaml:22]: Is it a good idea to upload all versions? (snapshot and release) @micedre

**Review by mboisnard — CHANGES_REQUESTED** [review]
Hi, 

Thanks for this PR ! Could you please rebase your commits and improve your commit message? DCO bot has failed due to signed-off rule (please use: git commit -s -m 'Ceci est mon message de validation').

Some points in review and below:
- we can remove .travis.yml file now
- we can use the Cache Action to locally cache maven dependencies (https://github.com/actions/cache)
- could you please add a line break after each step to beautify this workflow? ^^

File: .github/workflows/tag.yaml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Set up JDK 11
```
Comment (mboisnard) [inline .github/workflows/tag.yaml:22]: jdk 8

File: .github/workflows/workflows.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Set up JDK 11
```
Comment (mboisnard) [inline .github/workflows/workflows.yaml:25]: jdk 8

File: .github/workflows/workflows.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
```
Comment (mboisnard) [inline .github/workflows/workflows.yaml:28]: java-version 8

File: .github/workflows/tag.yaml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    steps:
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::$(echo $GITHUB_REF | cut -d / -f 3)
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Checkout artifact
+        id: download
```
Comment (mboisnard) [inline .github/workflows/tag.yaml:52]: Is id mandatory here?

File: .github/workflows/tag.yaml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: ${{ github.ref }}
+          body: |
+            ${{github.event.head_commit.message}}
+          draft: false
+          prerelease: false
+
+      - name: Upload Release javadoc
+        id: upload-release-javadoc
```
Comment (mboisnard) [inline .github/workflows/tag.yaml:72]: Is id mandatory here?

File: .github/workflows/tag.yaml [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
+          asset_path: ./target/keycloak-franceconnect-${{ steps.get_version.outputs.VERSION }}-javadoc.jar
+          asset_name: keycloak-franceconnect-${{ steps.get_version.outputs.VERSION }}-javadoc.jar
+          asset_content_type: jar
+
+      - name: Upload Release sources
+        id: upload-release-sources
```
Comment (mboisnard) [inline .github/workflows/tag.yaml:83]: Is id mandatory here?

File: .github/workflows/tag.yaml [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
+          asset_path: ./target/keycloak-franceconnect-${{ steps.get_version.outputs.VERSION }}-sources.jar
+          asset_name: keycloak-franceconnect-${{ steps.get_version.outputs.VERSION }}-sources.jar
+          asset_content_type: jar
+
+      - name: Upload Release
+        id: upload-release
```
Comment (mboisnard) [inline .github/workflows/tag.yaml:94]: Is id mandatory here?

**Review by mboisnard — CHANGES_REQUESTED** [review]

File: .github/workflows/workflows.yaml (hunk 2 above)
Comment (micedre) [reply .github/workflows/workflows.yaml:36]: I don't think we need to upload anything here.

File: .github/workflows/workflows.yaml [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+      - name: Set current version
+        run: mvn -B versions:set -DnewVersion=${GITHUB_REF##*/}
```
Comment (micedre) [inline .github/workflows/workflows.yaml:31]: Why do you change the version here ?

Comment (Donatien26) [reply .github/workflows/workflows.yaml:31]: Indeed it's useless, with this job we only want to see if maven build work. I remove it.

**Comment by micedre** [comment]
lgtm, @mboisnard, what do you think?

**Comment by mboisnard** [comment]
@micedre seems good to me.
Just a little question: is it good to have a different filename than the release name?
- filename uses $(echo $GITHUB_REF | cut -d / -f 3)  ==> aka tagName: tagPart1/tagPart2/tagPart3 (is this case is possible/useful??), filename: keycloak-franceconnect-tagPart1.jar
- release tagName & releaseName : ${{ github.ref }} ==> aka tagPart1/tagPart2/tagPart3 in my example
Do you really need to "cut -d / -f 3" with your tag convention?

File: .github/workflows/tag.yaml [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+
+      - name: Set current version
+        run: mvn -B versions:set -DnewVersion=${GITHUB_REF##*/}
```
Comment (mboisnard) [inline .github/workflows/tag.yaml:28]: Use project maven wrapper ./mvnw

File: .github/workflows/tag.yaml [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+      - name: Build with Maven
+        run: mvn -B clean verify
```
Comment (mboisnard) [inline .github/workflows/tag.yaml:31]: Use project maven wrapper ./mvnw

File: .github/workflows/workflows.yaml [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+
+      - name: Build with Maven
+        run: mvn -B clean verify
```
Comment (mboisnard) [inline .github/workflows/workflows.yaml:31]: Use project maven wrapper ./mvnw

**Review by mboisnard — CHANGES_REQUESTED** [review]
Just use project maven wrapper instead of unknown maven version inside the ubuntu-latest image

**Comment by micedre** [comment]
> 
> 
> @micedre seems good to me.
> Just a little question: is it good to have a different filename than the release name?
> 
>     * filename uses $(echo $GITHUB_REF | cut -d / -f 3)  ==> aka tagName: tagPart1/tagPart2/tagPart3 (is this case possible/useful??), filename: keycloak-franceconnect-tagPart1.jar
> 
>     * release tagName & releaseName : ${{ github.ref }} ==> aka tagPart1/tagPart2/tagPart3 in my example
>       Do you really need to "cut -d / -f 3" with your tag convention?

GITHUB_REF is something like "refs/tags/v1.2.3", we just want the last part I think (but for both, maybe we can set it globally, to use it everywhere, @Donatien26 , can ou look into it ?)

**Comment by micedre** [comment]
I'm merging this, we can always modify it later.

**Pull request merged**